addon/mixins/relationship-support-serializer.js

```diff
@@ -0,0 +1,40 @@
+import getAdapterOption from '@bagaar/ember-data-bagaaravel/utils/get-adapter-option';
+import getRelationshipDescriptor from '@bagaar/ember-data-bagaaravel/utils/get-relationship-descriptor';
+import Mixin from '@ember/object/mixin';
+
+export default Mixin.create({
+  /**
+   * Hooks
+   */
+
+  serialize(snapshot) {
+    let serialized = this._super(...arguments);
+    let relationshipName = getAdapterOption(snapshot, 'relationshipName');
+
+    if (!relationshipName) {
+      return serialized;
+    }
+
+    let serializedRelationships = serialized.data.relationships;
+
+    if (serializedRelationships && serializedRelationships[relationshipName]) {
+      return serializedRelationships[relationshipName];
+    }
+
+    let relationshipDescriptor = getRelationshipDescriptor(snapshot.record, relationshipName);
+
+    if (relationshipDescriptor.kind === 'hasMany') {
```

Comment:
Would be nice if Ember Data exported some "types" for this. Maybe we can const this ourselves? or create some util check functions (for ex. `isHasManyRelationship(kind)`) but that might be overkill.

Reply:
We have an `isHasManyRelationship` util in the old version. Since this is the only use case, I thought it would be overkill. Could use a const though?

Reply:
Nah, it's probably overkill, nevermind.